src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -155,12 +151,36 @@ public void DeleteUpdatableWorkloadsFile()
             }
         }
 
-        public static string WorkloadSetVersionToWorkloadSetPackageVersion(string setVersion)
+        public static string WorkloadSetVersionToWorkloadSetPackageVersion(string setVersion, out SdkFeatureBand sdkFeatureBand)
         {
-            var nugetVersion = new NuGetVersion(setVersion);
-            var patch = nugetVersion.Revision;
-            var release = string.IsNullOrWhiteSpace(nugetVersion.Release) ? string.Empty : $"-{nugetVersion.Release}";
-            return $"{nugetVersion.Major}.{nugetVersion.Patch}.{patch}{release}";
+            string[] sections = setVersion.Split(new char[] { '-', '+' }, 2);
+            string versionCore = sections[0];
+            string preReleaseOrBuild = sections.Length > 1 ? sections[1] : null;
+
```

Comment:
Why are you doing all this manually instead of using NugetVersion's parsing? I'm having a little trouble figuring out what exactly you're doing here.

/cc: @joeloff